lib/modules/manager/devcontainer/extract.spec.ts

```diff
@@ -0,0 +1,363 @@
+import upath from 'upath';
+import * as dockerfileExtract from '../dockerfile/extract';
+import { extractPackageFile } from '.';
+
+function getFixture(content: string) {
+  return {
+    content,
+    path: upath.resolve(__dirname, 'devcontainer.json'),
+  };
+}
+
+describe('modules/manager/devcontainer/extract', () => {
+  describe('extractPackageFile()', () => {
+    afterEach(() => {
+      jest.restoreAllMocks();
+    });
+
+    it('returns null when the dev container JSON file is empty', () => {
+      // Arrange
+      const content = '';
+      const packageFile = '';
+      const extractConfig = {};
+      // Act
+      const result = extractPackageFile(content, packageFile, extractConfig);
+
+      // Assert
+      expect(result).toBeNull();
+    });
+
+    it('returns null when the dev container JSON file contents are malformed', () => {
+      // Arrange
+      const content = 'malformed json}}}';
+      const packageFile = '';
+      const extractConfig = {};
+      // Act
+      const result = extractPackageFile(content, packageFile, extractConfig);
+
+      // Assert
+      expect(result).toBeNull();
+    });
+
+    it('returns feature image deps when only the features property is defined in dev container JSON file', () => {
+      // Arrange
+      const fixture = getFixture(`
+      {
+        "features": {
+          "devcontainer.registry.renovate.com/test/features/first:1.2.3": {},
+          "devcontainer.registry.renovate.com/test/features/second:4.5.6": {}
+        }
+      }`);
+      const extractConfig = {};
+      // Act
+      const result = extractPackageFile(
+        fixture.content,
+        fixture.path,
+        extractConfig,
+      );
+
+      // Assert
+      expect(result).not.toBeNull();
+      expect(result?.deps).not.toBeNull();
+      expect(result?.deps.length).toBe(2);
+      expect(result?.deps).toMatchInlineSnapshot(`
+      [
+        {
+          "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
+          "currentDigest": undefined,
+          "currentValue": "1.2.3",
+          "datasource": "docker",
+          "depName": "devcontainer.registry.renovate.com/test/features/first",
+          "replaceString": "devcontainer.registry.renovate.com/test/features/first:1.2.3",
+        },
+        {
+          "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
+          "currentDigest": undefined,
+          "currentValue": "4.5.6",
+          "datasource": "docker",
+          "depName": "devcontainer.registry.renovate.com/test/features/second",
+          "replaceString": "devcontainer.registry.renovate.com/test/features/second:4.5.6",
+        },
+      ]
+      `);
```

Comment:
use single `toEqual`

Reply:
I've removed the initial assertions but changing `toMatchInlineSnapshot` to `toEqual` gets automatically changed to `toBe` and then fails because of the JSON formatting, so I've left it using `toMatchInlineSnapshot`.